Gameplay/RoomTeleporter.cpp

```diff
@@ -0,0 +1,162 @@
+#include "scriptingUtil/gameplaypch.h"
+#include "RoomTeleporter.h"
+
+// NOTE FOR BERNAT: To scripts be able to work, you need to define it's 
+// constructor properly. Note the name field that is same with the class name.
+Hachiko::Scripting::RoomTeleporter::RoomTeleporter(GameObject* game_object)
+	: Script(game_object, "RoomTeleporter")
+	, _player(nullptr)
+	, _room_portal()
+	, _outdoor_portal(nullptr)
+	, _fade_image(nullptr)
+	, _fade_duration(0.25f)
+	, _trigger_distance(1.f)
+{
+}
+
+void Hachiko::Scripting::RoomTeleporter::OnAwake()
+{
+
+}
+
+void Hachiko::Scripting::RoomTeleporter::OnStart()
+{
+	// Don't forget, if you change any field here, it will override whatever 
+	// you set from the editor, as OnLoad is called before OnAwake. As opposed
+	// to OnStart, OnAwake is called again when you build scripts again at run
+	// time.
+
+	if (!_fade_image) return;
+
+	_image_transform = _fade_image->GetGameObject()->GetComponent<ComponentTransform2D>();
+
+	if (!_image_transform) return;
+
+	// Center image on the screen
+	_image_transform->SetAnchor(float2(0.5));
+
+	// Make image occupy entire screen (Scale multiplies final size)
+	_image_transform->SetSize(float2(10.f));
+	_image_transform->SetScale(float2(25000.f));
+
+	// Set initial color to black without alpha
+	_fade_image->SetColor(_clear_color);
+}
+
+void Hachiko::Scripting::RoomTeleporter::OnUpdate()
+{
+	if (!_player || !_outdoor_portal || !_room_portal || !_fade_image)
+	{
+		// If some reference is working the system cant work
+		return;
+	}
+	
+	if(Input::IsKeyDown(Input::KeyCode::KEY_T))
```

Comment:
The key T it's for the interaction with the player, for trigger the scene change?